src/realm/collection.hpp

```diff
@@ -765,6 +964,26 @@ struct CollectionIterator {
     size_t m_ndx = size_t(-1);
 };
 
+template <class T>
+class IteratorAdapter {
```

Comment:
TODO: no so sure this wrapper is really needed. Also, we should avoid doing `static_cast` in order to iterate through the BST, probably a better design could be to return the proper iterator via the `BPlusTree<T>`